coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -823,6 +823,58 @@ def adaptive_avg_pool2d(context, node):
 
     context.add(avg_pool)
 
+@register_torch_op
+def adaptive_max_pool2d(context, node):
+    inputs = _get_inputs(context, node, expected=2)
+
+    _input = inputs[0]
+    output_size = inputs[1].val
+    assert isinstance(output_size, _np.ndarray)
+    output_size = tuple(output_size)
+
+    if output_size == (1, 1):
+        # Represent (1,1) output size via @reduce_max
+        # Assume channel first ordering, reduce the last two (HW) dims.
+        axes = mb.const(val=[-2, -1], name=node.name + "_axes")
+        keep_dims = mb.const(val=True, name=node.name + "_keep_dims")
+
+        max_pool = mb.reduce_max(
+            x=_input, axes=axes, keep_dims=keep_dims, name=node.name
+        )
+    elif _input.shape is not None:
+        # TODO: The calculations to convert adaptive_pool to standard pool,
+        # given a known input size, come from
+        # https://stackoverflow.com/questions/53841509/how-does-adaptive-pooling-in-pytorch-work
+        # However, as indicated in that SO, this isn't quite how PyTorch
+        # computes adaptive pooling, leading to inaccuracies in model outputs.
+        # rdar://60900834
+        strides = [ind // outd for ind, outd in zip(_input.shape[-2:], output_size)]
+        pad_type = "valid"
+        # Need to explicity state L-R, T-B pad
+        pad = [0, 0, 0, 0]
+        dilation = [1, 1]
+        kernel_sizes = [
+            ind - s * (outd - 1)
+            for ind, outd, s in zip(_input.shape[-2:], output_size, strides)
+        ]
+        max_pool = mb.max_pool(
+            x=_input,
+            kernel_sizes=kernel_sizes,
+            strides=strides,
+            pad_type=pad_type,
+            pad=pad,
+            name=node.name,
+        )
+    else:
+        raise ValueError(
+            "adaptive_max_pool2d only supported when input tensor size is known or output size == (1,1). Recived: input size == {}, output size == {}".format(
+                _input.shape_str(), output_size,
+            )
+        )
+
+    context.add(max_pool)
```

Comment:
Can you add some test cases in 

https://github.com/apple/coremltools/blob/master/coremltools/converters/mil/frontend/torch/test/test_torch_ops.py

You can include both tests that passes and also tests that `xfails` . See xfail example https://github.com/apple/coremltools/blob/master/coremltools/converters/mil/frontend/torch/test/test_torch_ops.py#L824

That way it's more clear to what extent can we support adaptive pool op.

Reply:
To run those tests you just need to do `pytest /path/to/test_torch_ops.py::TestClassName` etc (you might need to pip install a few dependencies)

Reply:
I've started work on the unit test but I don't know if it's working yet. I have not been able to run the existing tests successfully, because the test runner skips all tests. I think that's because I'm running Python 3.8 (which seg faults on tests according to a note in the source) and/or because I need to compile a local version of the library (to import CoreML framework) and then pip install that as an editable library in order for tests to run successfully. I'll have a look.

Reply:
I have added tests which include cases with expected success (where the input and output sizes are such that coremltools can convert it correctly) and expected failure (where this is not so).

I parameterized a test argument which produces variation between the two cases. I'm new to this testing approach so please do let me know if something is off.  @daiwei89